packages/components/src/toggle-group-control/test/index.js

```diff
@@ -7,24 +7,52 @@ import { render, fireEvent, screen } from '@testing-library/react';
  * Internal dependencies
  */
 import { ToggleGroupControl, ToggleGroupControlOption } from '../index';
+import { TOOLTIP_DELAY } from '../../tooltip';
 
 describe( 'ToggleGroupControl', () => {
+	beforeEach( () => {
+		jest.useFakeTimers();
+	} );
+
+	afterEach( () => {
+		jest.useRealTimers();
```

Comment:
The [docs from React Testing Library](https://testing-library.com/docs/using-fake-timers/) suggest to also add 

```suggestion
		jest.runOnlyPendingTimers();
		jest.useRealTimers();
```

Reply:
Thanks for the suggestion.

The docs are ambiguous on this one I see. See https://jestjs.io/docs/timer-mocks#run-pending-timers, which loosely recommends running `jest.runOnlyPendingTimers();` to clear regular intervals.  

I understand that we'd want to flush out pending timers though, and I did try using `runOnlyPendingTimers` originally but it caused the tests to fail, and I'm not really sure why. 

This was also after trying to remove `setTimeout` from the test itself and use `jest.advanceTimersByTime( TOOLTIP_DELAY );` and `runOnlyPendingTimers` within the test. `jest.clearAllTimers()` in the `afterEach` worked as a fallback.

`<Tooltip />` uses `useDebounce` so it might be due to me not testing correctly.

🎉 🎉 🎉 🎉 🎉 🎉 🎉 

Anyway, there's a happy ending. It turns out that `<Tooltip />` only debounces `onMouseEnter` and not `focus` so I switched to testing using that event. 

It's cleaner, and with no `setTimeout`, no fake timers, and `<Tooltip />` already has test coverage for `mouseEnter|Leave`.  

WOOT!

Reply:
That's great, thank you@